www/js/controllers.js

```diff
@@ -48,55 +47,94 @@ app.controller('HistoryController',function($rootScope, $scope, Auth, UserFinish
     $scope.authUser = Auth.$getAuth();
     $scope.finishedItems = UserFinishedTasks($scope.authUser.uid);
     $scope.clearAll = function(){
-       // finishedItemsRef.remove();
-        console.log(">>>> "+UserFinishedTasks($scope.authUser.uid).length);
+        console.log(">>>> " +UserFinishedTasks($scope.authUser.uid).length);
     }
 });
-app.controller('ToDoListActiveController', function ($rootScope, $scope, $ionicModal, Auth, UserTasks,UserFinishedTasks,$ionicListDelegate) {
-    $scope.authUser = Auth.$getAuth();
-    $scope.items = UserTasks($scope.authUser.uid);
-    $scope.finishedItems = UserFinishedTasks($scope.authUser.uid);
+app.controller('TaskController', function($scope, $state, $stateParams, $ionicHistory,
+                                          UserTask, UserTasks, Auth, $cordovaDatePicker, $interval,
+                                          $ionicListDelegate) {
 
+    $scope.authUser = Auth.$getAuth();
+    $scope.isNewTask = $stateParams.taskId == 'new';
+    $scope.username = getDisplayNameFromAuth(Auth.$getAuth()).split(" ").shift();
+    var ownerId = $stateParams.ownerId;
+    if (!ownerId) {
+        ownerId = $scope.authUser.uid;
+    }
+    $scope.setTimer = function(task){
+        if(task.dueDate == 0){
 
-    $scope.showEditTask = function (item) {
-        $scope.isNewTask = item == undefined;
-        $scope.task = item ? item : {} ;
-        $scope.modal.show();
+        } else {
+            $scope.dueTime = task.dueDate - (new Date().getTime());
+            var timer = $interval( function(){
+                if($scope.dueTime == 0){
+                    $interval.cancel(timer);
+                }
+                $scope.dueTime = $scope.dueTime - 1000;
+            }, 1000);
+        }
     };
-
-    $scope.addTask = function (task) {
-        if (task.message.trim() == '') {
+    if ($scope.isNewTask) {
+        var createdDate = new  Date();
+        $scope.task = {
+            ownerId: ownerId,
+            ownerName: getDisplayNameFromAuth(Auth.$getAuth()),
+            createdAt: createdDate.getTime(),
+            dueDate : 0
+        }
+    } else {
+        $scope.task = UserTask(ownerId, $stateParams.taskId);
+        $scope.setTimer($scope.task);
+    }
+    $scope.saveTask = function (task) {
+        function failureCallback(error) {
             // TODO Show error.
+        }
+        if ( task == undefined || task.message.trim() == '') {
+            failureCallback();
             return;
         }
-        $scope.items.$add({message: task.message}).then(function(ref) {
-            console.log(">> Successfully saved a message ", ref.key(), ref.$id);
-            task.message = '';
-            $scope.closeNewTask();
-        }).catch(function(error) {
-            // TODO show error on the modal itself.
-        });
+        function successCallback(ref) {
+            $ionicHistory.goBack();
```

Comment:
actually, not in successCallback, it should be done in `timeReminder`. but why did you do it in `catch` block ?